script.js

```diff
@@ -1 +1,519 @@
-console.log('yay!');
+// Global variables
+const board = [];
+
+// To track position of cards clicked and matched
+const positionArray = [];
+let firstCol;
+let firstRow;
+
+let firstCard = null;
+let secondCard = null;
+
+// Global state to track if cards are currently matched - purpose is to prevent the 3rd click which  yields a new firstCardHtml from erasing the 1st click's firstCardHtml
+let currentNotMatched = true;
+// track the first card and second card's HTML equivalent
+let firstCardHtml;
+let secondCardHtml;
+const countDownTime = 180; // 180s
+
+let ref = ''; // define & capture reference to timer event;
+let ref2 = ''; // define & capture reference to the event where first 3 cards are clicked and the 1st 2 cards must be cleared.
+let ref3 = ''; // special event
+let timeLeft = 0; // capture the amount of time left; initialized to 0
+
+const boardSize = 4; // has to be an even number
+// game only starts when squareClicked.
+let numOfClicks = 0;
+let score = 0;
+let deck;
+
+// Track position of card clicked and prevent user from clicking on the same card multiple times to score
+const positionOfCardClicked = {
+  column: '',
+  row: '',
+};
+// create resetButton here
+const resetButton = document.createElement('button');
+resetButton.setAttribute('id', 'rest');
+
+// Main Output Area:
+const outputValue = 'Hi, please input your name :)';
+const outputDivTag = document.createElement('div');
+
+// Score Output Area:
+const scoreOutputDivTag = document.createElement('div');
+
+// Time Output Area;
+const timer = document.createElement('div');
```

Comment:
Good to declare these globals at the top, but consider only doing the initialisation in `gameInit`, where the other DOM element initialisation is done, so that more of the DOM initialisation can be done in the same place for clarity.